lib/components/MusicScreen/music_screen_tab_view.dart

```diff
@@ -185,42 +185,53 @@ class _MusicScreenTabViewState extends State<MusicScreenTabView>
   // If clicked in the # element, it goes to the first one ( pixels = 0 )
   void scrollToLetter(String? clickedLetter) async {
     String? letter = clickedLetter ?? letterToSearch;
-    bool reversed = FinampSettingsHelper
-            .finampSettings.tabSortOrder[widget.tabContentType] ==
-        SortOrder.descending;
     if (letter == null || letter.isEmpty) return;
 
     letterToSearch = letter;
-    var letterCodePoint = letterToSearch!.toLowerCase().codeUnitAt(0);
+    var letterToScrollTo = letterToSearch!.toLowerCase()[0];
+
+    const scrollDuration = Duration(milliseconds: 750);
 
     if (letter == '#') {
-      letterCodePoint = 0;
-    }
-    for (var i = 0; i < _pagingController.itemList!.length; i++) {
-      var itemCodePoint =
-          _pagingController.itemList![i].nameForSorting!.codeUnitAt(0);
-      var diff = itemCodePoint - letterCodePoint;
-      if (reversed ? diff <= 0 : diff >= 0) {
-        timer?.cancel();
-        if (reversed ? diff < 0 : diff > 0) {
-          await controller.scrollToIndex(i,
-              duration: const Duration(milliseconds: 200),
-              preferPosition: AutoScrollPosition.middle);
-        } else {
+      await controller.scrollToIndex(0,
+        duration: scrollDuration,
+        preferPosition: AutoScrollPosition.begin);
+      return;
+    } else {
+      //TODO use binary search to improve performance for already loaded pages
+      bool reversed = FinampSettingsHelper.finampSettings.tabSortOrder[widget.tabContentType] == SortOrder.descending;
+      for (var i = 0; i < _pagingController.itemList!.length; i++) {
+        var itemFirstLetter =
+            _pagingController.itemList![i].nameForSorting![0];
+        final comparisonResult = letterToScrollTo.compareTo(itemFirstLetter);
+        if (comparisonResult == 0) {
+          timer?.cancel();
           await controller.scrollToIndex(i,
-              duration: const Duration(milliseconds: 200),
+              duration: scrollDuration,
+              preferPosition: AutoScrollPosition.begin);
+          
+          letterToSearch = null;
+          return;
+        } else if (reversed ? comparisonResult > 0 : comparisonResult < 0) {
+          // If the letter is before the current item, there was no previous match (letter doesn't seem to exist in library)
+          // scroll to the previous item instead
+          timer?.cancel();
+          await controller.scrollToIndex((i - 1).clamp(0, (_pagingController.itemList?.length ?? 1) - 1),
+              duration: scrollDuration,
               preferPosition: AutoScrollPosition.begin);
```

Comment:
I actually couldn't tell a difference when testing. But maybe that was due to some other things I messed up...

Reply:
The goal is to put the place where the letter would be in the middle of the screen when we fail to find a letter, so that it's clear the letter doesn't exist.   Doing this is an alternative to backing the target index up by X items.  It clearly changes the final position whenever I try it.

Reply:
I guess that makes sense. Sorry for being dense, I'll revert it then...

Reply:
So the reason I didn't notice is because the only missing letter on my test server is `X`, and there are only a few items with `Y` and `Z`, so it just never ended up scrolling past the middle anyway...